nixio/dimensions.py

```diff
@@ -313,3 +314,78 @@ def labels(self):
     def labels(self, labels):
         dt = util.vlen_str_dtype
         self._h5group.write_data("labels", labels, dtype=dt)
+
+
+class DataFrameDimension(Dimension):
+
+    def __init__(self, h5group, index):
+        super(DataFrameDimension, self).__init__(h5group, index)
+
+    @classmethod
+    def _create_new(cls, parent, index, data_frame, column):
+        """
+        Create new Dimension that point to a DataFrame
+        :param parent: the linked DataArray
+        :param parent: the linked DataArray
+        :param data_frame: the DataFrame that
+        :param column: optional, the column that it points to
+        :return:
+        """
+        newdim = super(DataFrameDimension, cls)._create_new(parent, index)
+        newdim.data_frame = data_frame
+        newdim.column = column
+        newdim._dimension_type = DimensionType.DataFrame
+        return newdim
+
+    def unit(self, index=None):
+        if index is None:
+            if self.column is None:
+                raise ValueError("No default column index is set for this Dimension. Please supply one")
+            else:
+                idx = self.column
+        else:
+            idx = index
+        unit = self.data_frame.units[idx]
+        return unit
+
+    def ticks(self, index=None):
+        if index is None:
+            if self.column is None:
+                raise ValueError("No default column index is set for this Dimension. Please supply one")
+            else:
+                idx = self.column
+        else:
+            idx = index
+        df = self.data_frame
+        ticks = df[df.column_names[idx]]
+        return ticks
+
+    def label(self, index=None):
+        if index is None:
+            if self.column is None:
+                label = self.data_frame.name
+            else:
+                label = self.data_frame.column_names[self.column]
+        else:
+            label = self.data_frame.column_names[index]
+        return label
+
+    @property
+    def data_frame(self):
+        dfname = self._h5group.get_by_pos(0).name
+        grp = self._h5group.open_group(dfname)
+        df = DataFrame(grp.h5root.group["data_frames"], grp)
+        return df
+
+    @data_frame.setter
+    def data_frame(self, df):
+        self._h5group.create_link(df, df.id)
```

Comment:
```suggestion
        self._h5group.create_link(df, "dataframe")
```

I think it might be simpler if the link to the DataFrame from the DataFrameDimension has a fixed name (e.g., `"dataframe"`), instead of linking by ID.  Just like with Features, Positions, and Extents, this would make it clearer what the purpose of the link is in the file and would make retrieving the referenced object simpler (see the `data_frame` property getter above).

@jgrewe Thoughts?